app/client-v2/setupTests.ts

```diff
@@ -2,7 +2,7 @@ import { server } from "@/mocks/server";
 import "@testing-library/jest-dom/vitest";
 
 beforeAll(() => {
-  server.listen();
+  server.listen({ onUnhandledRequest: "error" });
```

Comment:
Maybe better to have another place they might be caught just in case? That's my vote, will wait and see how reviewer 2 votes.

Reply:
Possibly, someone will introduce something later that doesn't (yet) use a default handler. I can see how it's a bit duplicatey, though.

Reply:
Yeah, the default handler will in fact handle everything and return an empty success response so `onUnhandledRequest` will never be triggered, so there isn't a lot of point to this change.

I suppose there's an argument to had about whether it would be better to omit the default handler and require the tests to explicitly handle all expected requests. But the handlers are quite disconnected from the individual tests so they're not really part of the testing of expected behaviour, it's just potentially slightly confusing when odd behaviour happens in the test because you forgot the code was expecting a response from some endpoint. But that's a separate issue!